[11.0][FIX] frontend tests

kirca opened the pull request:
Test `point_of_sale` in travis to check if there is issue with phantomjs. Related to https://github.com/OCA/pos/issues/309

chienandalu commented:
@kirca I was testing this locally and found out that the problem seems to rely on the interaction with this part of `pos_lot_selection` code: https://github.com/OCA/pos/blob/11.0/pos_lot_selection/static/src/js/chrome.js#L10-L37

It's a part that injects some code into the instanced widget of lot input. I can't guess where the problem relies though. Yo can raise the error locally running the two modules tests altogether.

kirca commented:
@chienandalu , thanks. I'll check if I can reproduce it and if it is possible update the test case so that it doesn't break.

chienandalu commented:
Ok, let's keep this PR open to try to greenify the branch :wink:

kirca commented:
@chienandalu , it is finally :green_apple:

chienandalu reviewed (commented):
Was it en ES6 support thing from phantomjs all along?

kirca commented:
@chienandalu , that is right. It is explained in the commit message.

chienandalu commented:
Great! Thanks!

pedrobaeza commented on pos_lot_selection/__manifest__.py:
```diff
@@ -6,6 +6,7 @@
     'version': '11.0.1.0.0',
     'category': 'Point of Sale',
     'author': 'Tecnativa,'
+              'Lambda IS,'
```
Sorry, but this change doesn't grant you co-authorship in the module. Please add yourself only in contributors.

kirca replied:
Sure. Just curious, what guidelines do you follow on what qualifies as a change for authorship?

pedrobaeza replied:
There are no objective metrics, but a matter of sense about the amount of time/code involved around the module itself. In this case, it's very thankfully that you have fixed a problem in a module that is not yours, but the module essence and the amount of work invested for developing it is not something that you have needed to invest. Don't you think?

kirca replied:
My opinion is that anyone who modifies a file is a co-author and has copyright on those files. If order is preserved (top to bottom) it can be seen who is original author. Qualifying based on amount of work done is quite subjective and if it will be used as a criteria we should define it in the [guidelines](https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst), example _only improvements grant co-authorship and can be listed in the manifest, authors of bugfixes and other changes can be added in contributors_. I'll remove myself from the authors now although I did spend a surprising amount of time debugging and researching this :wink: .

pedrobaeza replied:
One think is the copyright of the file, and another the module co-authorship. If you have a proposal for "objectivitizing" the module co-authorship in the guidelines, please do it. I'm all opened for avoiding these misunderstandings.

kirca replied:
OK, I'll see if I can draft a proposal. Thanks for the review.

kirca committed:
[FIX] pos_lot_selection: ES6 code breaks phantomjs tests

Object.assign is introduced in ES6 (2015) which is not supported by
PhantomJS. Replace it with the equivalent _.extend function to prevent
frontend tests from failing.

1. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign#Specifications
2. https://github.com/ariya/phantomjs/issues/14506
